### ISSUE TITLE
cpp: force static functions

### ISSUE THREAD

**Issue by ftynse** [PollyLabs/isl#22]
Currently, the exported function is made static if its first argument is of a different type than its expected surrounding class (determined from the function name prefix).

There are "constructing" functions which make more sense as static rather than as instance functions, but with current generator it seems impossible to make them static. In particular, space construction:
`isl_space_map_from_domain_and_range(isl_space *, isl_space *)` makes more sense if translated as
`isl::space::map_from_domain_and_range(domain, range)` and not as `domain.map_from_domain_and_range(range)`.  Same goes for, e.g., `isl_space_set_from_params`.

I would suggest an extra tag `__isl_export_ensure_static` that forces the generator to produce a static function even if the function would be instance function otherwise. The danger here is to avoid a mixup with all the implicitly static functions with only `__isl_export`.

**Comment by tobiasgrosser** [comment]
Good point. I have seen it also for isl::space::map_from_domain_and_range.

An alternative name could be:

__isl_constructor_static

**Comment by ftynse** [comment]
I was thinking about that name but wouldn't that require to replace all 
already existing `__isl_export` that actually create a static 
constructor with `__isl_constructor_static` ?

**Comment by tobiasgrosser** [comment]
Why? If you want to make it more explicit, what about:

__isl_constructor_ensure_static

I think the best is to submit a patch to isl and discuss with Sven. No need to have two bikesheding discussions.
